### DISCUSSION TITLE
[Code Quality]: Tooling & .NET 4.8 migration

### DISCUSSION THREAD

**Discussion by unquietwiki** [d2phap/ImageGlass#773] [General] [answered]
https://github.com/unquietwiki/ImageGlass/tree/develop

Since I've used this program for 2-3 years now and was just looking at another program I use regularly, I figured I'd take a crack at making some codebase improvements. Here's what I'm going for...

- [.NET 4.8](https://dotnet.microsoft.com/download/dotnet-framework/net48) as a target runtime in the Solution components.
- Use [CodeMaid](http://www.codemaid.net/) to enforce formatting cleanup.
- Use [Sharpen](https://sharpen.rocks/) to suggest refactorings.

Sharpen can't yet do its refactorings automatically; it leans on user interaction + VS Code's own refactoring suggestions. The C# 5.0 & 8.0 refactorings are more involved because they involve async & null changes; the rest involve more basic changes, that have a less disruptive impact. I've done some of the work, but there remains much to do. I have other stuff I need to work on, so I don't want to put too much time into this if it's not going to be of use. Please advise. Thanks!

**Comment by unquietwiki** [comment]
@d2phap BTW thank you for answering my previous question a while back about adding a codec. Since this is your work, I'd like your thoughts on this.

**Comment by d2phap** [comment]
Hi @unquietwiki 

It's good to know you want to refactor and clean up the code.

Actually, I have plan to move IG from WinForms to WPF with .NET 5; this will take long time!
The WPF version will reuse most of the code, except for `ImageGlass.ImageBox`, `ImageGlass.ImageListView`, and the UI code for WinForms...

Please feel free to create small pull requests of refactoring. I'm happy to merge it.
Thanks a lot!

**Reply by unquietwiki** [reply]
@d2phap Thanks! I'll pull the first set now for your consideration.

**Reply by unquietwiki** [reply]
#775

**Reply by d2phap** [reply]
Sorry, I didn't mention that there is config file for formatting at  https://github.com/d2phap/ImageGlass/blob/master/Source/.editorconfig

and the external projects should not be cleaned up

**Comment by unquietwiki** [comment]
Acknowledged. Are you able to try the tools I'm using with your current
workflow & can compare results? I'm probably not going to get anything done
on this for another 12-24hr.

**Reply by d2phap** [reply]
Thank you, I mostly work on IG during weekend

**Comment by unquietwiki** [comment]
Hey @d2phap . I'll close #775 & try again to lay down basic refactoring, based on the following...

- Your editor config; no use of CodeMaid.
- [.NET 4.8](https://dotnet.microsoft.com/download/dotnet-framework/net48) as a target runtime in the Solution components, with auto-resolve of dependencies.
- [Sharpen](https://sharpen.rocks/) for identifying obsolete code chunks.
- I'll be using [Codacy](https://www.codacy.com/) on my repo to identify issues; you may want to as well (they're having a maintenance window this weekend, else they're pretty reliable).

**ImageGlass.ImageBox** and **ImageGlass.ImageListView**: you mentioned them as "external projects" you didn't want to mess with. I wasn't sure if they were in your repo, or not. How do you want to handle anything with those, that might be an issue?

**Reply by d2phap** [reply]
Hi @unquietwiki 

Thank you, I have been very busy with my current job.

- Codacy seems not free (or have free license for FOSS).

> ImageGlass.ImageBox and ImageGlass.ImageListView: you mentioned them as "external projects" you didn't want to mess with. I wasn't sure if they were in your repo, or not. How do you want to handle anything with those, that might be an issue?

These are the customised repos, and added in ImageGlass folder. I sometimes still look into the old source code to investigate somethings.

**Reply by unquietwiki** [reply]
@d2phap Acknowledged on the repos you identified. As for Codacy, they apparently support free use up to 4 users per https://www.codacy.com/pricing (I have it on for a couple of my own Github repos).

I hear you on the work; that's happened to me over the years. I don't even have a regular job ATM (obviously the Pandemic's a factor), but the family keeps me on my toes since they're still busy! Also looking into Javascript dev as well. Hope you're keeping well.

**Comment by unquietwiki** [comment, marked as answer]
@d2phap Hey there, hope things are going alright for you. I finally had a chance to do what we discussed and tried out the agreed-upon changes on the master branch. Seems to run pretty smoothly. Can you please let me know how you want this handled going forward? I imagine you'd want to have a branch set up to merge to, that we could finish up work on with your more recent updates? Thanks.

**Reply by unquietwiki** [reply]
https://github.com/unquietwiki/ImageGlass/tree/issue773

**Reply by d2phap** [reply]
Hey @unquietwiki 
I'm good! Yes, please create a pull request for this. I've been recently just exploring new things for IG, preparing for next version. The code in `develop` branch is the latest.
Thanks!

**Reply by unquietwiki** [reply]
@d2phap Thanks! https://github.com/d2phap/ImageGlass/pull/811 submitted for you to handle, as able. I need to get to bed; I'll follow-up later in the day as able. Glad to hear you're alright.

**Reply by d2phap** [reply]
Thanks a lot @unquietwiki 
Good night!

**Reply by unquietwiki** [reply]
Follow-up: noting for the record that #813 had additional refactoring completed.

**Reply by d2phap** [reply]
thanks a lot!

**Reply by unquietwiki** [reply]
@d2phap No problem! You won't believe how many random images + my own photography I have to manage. Heh.

**Reply by d2phap** [reply]
Don't say it's a million 😵

**Comment by unquietwiki** [comment]
@d2phap (counts) 11K in RAW + JPG for photos; another 11K+ in random crap I've found over the past 20 years. I used to run a BBS in High School when Internet was becoming a thing. Ah dial-up.
